app/controllers/concerns/integrator/hyrax/works_behavior.rb

```diff
@@ -68,6 +69,7 @@ def update_attributes
       end
 
       private
+        # @todo get this from a custom header or default to this
```

Comment:
Modify set_work_klass to read model from header 
```
        def set_work_klass
          if headers.fetch(:hyrax_work_model, nil)
            @work_klass = headers[:hyrax_work_model].constantize
          else
            @work_klass = WillowSword.config.work_models.first.constantize
          end
        end
```

Reply:
I've also added a check to make sure the specified model is in the work_models array.